src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -39,6 +30,72 @@ def child_sort_key(item):
     return [item["start_timestamp"], item["timestamp"]]
 
 
+def query_trace_data(trace_id, params):
+    transaction_query = discover.prepare_discover_query(
+        selected_columns=[
+            "id",
+            "transaction.status",
+            "transaction.op",
+            "transaction.duration",
+            "transaction",
+            "timestamp",
+            # project gets the slug, and project.id gets added automatically
+            "project",
+            "trace.span",
+            "trace.parent_span",
+            'to_other(trace.parent_span, "", 0, 1) AS root',
+        ],
+        # We want to guarantee at least getting the root, and hopefully events near it with timestamp
+        # id is just for consistent results
+        orderby=["-root", "-timestamp", "id"],
```

Comment:
Won't `-timestamp` be the most recent events first? I'm not super clear on why we're trying to get events around it though

Reply:
good catch, this should be `timestamp` the idea was that on traces with >100 transactions because events could be roughly sequential we'd get the earlier events in the trace so we could find their parent/child relationships

Reply:
Fixed in 50add9fb701dacef4bc1a8c2454155c263bf3347